src/PowerShellEditorServices/Extensions/FileContext.cs

```diff
@@ -239,6 +240,28 @@ public void Save()
             this.editorOperations.SaveFile(this.scriptFile.FilePath);
         }
 
+        /// <summary>
+        /// Save this file under a new path and open a new editor window on that file.
+        /// </summary>
+        /// <param name="newFilePath">
+        /// the path where the file should be saved,
+        /// including the file name with extension as the leaf
+        /// </param>
+        public void SaveAs(string newFilePath)
+        {
+            // Do some validation here so that we can provide a helpful error if the path won't work
+            string absolutePath = System.IO.Path.IsPathRooted(newFilePath) ?
+                newFilePath :
+                System.IO.Path.GetFullPath(System.IO.Path.Combine(System.IO.Path.GetDirectoryName(this.scriptFile.FilePath), newFilePath));
+
+            if (File.Exists(absolutePath))
+            {
+                throw new IOException(String.Format("The file '{0}' already exists", absolutePath));
```

Comment:
Does this crash pses or does it just throw an error in the PSIC? (hoping for the second)

Reply:
I think the second on the basis that it's a method on the `$psEditor` variable rather than in PSES itself (because I remember asking myself the same thing), but will check.